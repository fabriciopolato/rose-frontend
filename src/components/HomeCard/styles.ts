import styled from 'styled-components';

interface IProps {
  isReversed: boolean;
}

export const Container = styled.div<IProps>`
  width: 100%;
  padding: 0 10px;
  margin-top: 16px;

  & > article {
    width: 100%;
    height: 124px;

    display: flex;
    flex-direction: ${props => (props.isReversed ? 'row-reverse' : 'row')};
    justify-content: space-between;
    align-items: center;

    background: ${props => (props.isReversed ? props.theme.salmon : props.theme.orange)};
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    border-radius: 20px;

    & > figure {
      height: 100%;

      & > img {
        height: 100%;
      }
    }

    & > aside {
      height: 100%;
      padding: 16px;
      color: ${props => props.theme.black};
      text-align: ${props => (props.isReversed ? 'right' : 'left')};

      & > h3 {
        font-weight: 600;
        font-size: 1.4rem;
        margin-bottom: 8px;
      }

      & > p {
        font-weight: normal;
        font-size: 1.1rem;
      }
    }
  }
`;
