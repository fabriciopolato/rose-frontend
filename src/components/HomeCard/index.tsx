import React from 'react';

import { Container } from './styles';

interface IProps {
  isReversed?: boolean;
  image: string;
  title: string;
}

const HomeCard: React.FC<IProps> = ({
  isReversed = false, image, title, children,
}) => (
  <Container isReversed={isReversed}>
    <article data-testid="row">
      <figure>
        <img src={image} alt={title} />
      </figure>
      <aside>
        <h3>{title}</h3>
        <p>{children}</p>
      </aside>
    </article>
  </Container>
);

export default HomeCard;
