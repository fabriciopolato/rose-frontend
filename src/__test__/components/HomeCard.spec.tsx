import React from 'react';
import { render, getByAltText } from '@testing-library/react';

import HomeCard from '../../components/HomeCard';

describe('<HomeCard />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <HomeCard image="test.png" title="Title test">
        Paragraph test
      </HomeCard>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have image, title and paragraph provided by props', () => {
    const { getByText, getByAltText } = render(
      <HomeCard image="test.png" title="Title test">
        Paragraph test
      </HomeCard>
    );

    const paragraphElement = getByText('Paragraph test');
    const headingElement = getByText('Title test');
    const imageElement = getByAltText('Title test');

    expect(paragraphElement).toBeInTheDocument();
    expect(headingElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src', 'test.png');
  });

  it('should be reversed when isReversed is passed as a prop', () => {
    const { getByTestId } = render(
      <HomeCard isReversed image="test.png" title="Title test">
        Paragraph test
      </HomeCard>
    );

    const containerElement = getByTestId('row');

    expect(containerElement).toHaveStyle('flex-direction: row-reverse;');
  });
});
