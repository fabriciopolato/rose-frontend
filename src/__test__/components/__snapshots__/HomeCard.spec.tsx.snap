// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeCard /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jpTWJn"
  >
    <article
      data-testid="row"
    >
      <figure>
        <img
          alt="Title test"
          src="test.png"
        />
      </figure>
      <aside>
        <h3>
          Title test
        </h3>
        <p>
          Paragraph test
        </p>
      </aside>
    </article>
  </div>
</DocumentFragment>
`;
